deps/rabbit_common/src/mirrored_supervisor.erl

```diff
@@ -277,13 +277,16 @@ handle_call({init, Overall}, _From,
                            tx_fun             = TxFun,
                            initial_childspecs = ChildSpecs}) ->
     process_flag(trap_exit, true),
+    LockId = mirrored_supervisor_locks:lock(Group),
```

Comment:
Should the return value be checked here? If the `lock` didn't actually get lock in the expected 300s?

Reply:
It's a good point. `mirrored_supervisor:unlock/1` will be a no-op if the lock id is `undefined`.

Reply:
This is a convention we use on top of `global:set_lock/3` in peer discovery so migrated to Shovel and `mirrored_supervisor` as well.

Reply:
So it's not a mandatory lock, just some relaxed check. Probably the way the code is structured will never cause it fail - until one day it will. Maybe a warning should be logged in such a situation?

Reply:
Seconded. Since not having a lock can cause issues, we would like to know if users who run into issues have obtained a lock.

Reply:
Added a warning